pkgs/java_http/lib/src/java_client.dart

```diff
@@ -230,35 +231,58 @@ class JavaClient extends BaseClient {
     return contentLength;
   }
 
-  void _responseBody(
+  Future<void> _responseBody(
     Uri requestUrl,
     HttpURLConnection httpUrlConnection,
     SendPort sendPort,
     int? expectedBodyLength,
-  ) {
+  ) async {
     final responseCode = httpUrlConnection.getResponseCode();
 
-    final inputStream = (responseCode >= 200 && responseCode <= 299)
-        ? httpUrlConnection.getInputStream()
-        : httpUrlConnection.getErrorStream();
+    final responseBodyStream = (responseCode >= 200 && responseCode <= 299)
+        ? BufferedInputStream(httpUrlConnection.getInputStream())
+        : BufferedInputStream(httpUrlConnection.getErrorStream());
 
-    int byte;
     var actualBodyLength = 0;
-    // TODO: inputStream.read() could throw IOException.
-    while ((byte = inputStream.read()) != -1) {
-      sendPort.send([byte]);
-      actualBodyLength++;
+    final bytesBuffer = JArray(jbyte.type, 4096);
+    // TODO: read1() could throw IOException.
+    var bytesCount =
+        responseBodyStream.read1(bytesBuffer, 0, bytesBuffer.length);
+
+    while (bytesCount != -1) {
```

Comment:
This might be easier:

```dart
while (true) {
  final bytesCount = responseBodyStream.read1(bytesBuffer, 0, bytesBuffer.length); 
  if (bytesCount == -1) {
    break;
  }
      if (bytesCount == 0) {
        // No more data is available without blocking so give other Isolates an
        // opportunity to run.
        await Future<void>.delayed(Duration.zero);
        continue;
    }

   sendPort.send(bytesBuffer.toUint8List(length: bytesCount));
   actualBodyLength += bytesCount;
}

}

Reply:
Hey @brianquinlan,

I like your approach because it removes the duplication of:

```dart 
bytesCount = responseBodyStream.read1(bytesBuffer, 0, bytesBuffer.length);
```

Including a continue statement when bytesCount == 0 is a good idea.

I've pushed both of these changes :muscle:. 

Thank you!